src/conn-avatars.c

```diff
@@ -915,6 +933,8 @@ conn_avatars_fill_contact_attributes (GObject *obj,
 
           if (NULL != presence->avatar_sha1)
             g_value_set_string (val, presence->avatar_sha1);
+          else if (g_hash_table_lookup (self->pep_avatar_hashes,  GINT_TO_POINTER(handle)))
+            g_value_set_string (val, g_hash_table_lookup (self->pep_avatar_hashes,  GINT_TO_POINTER(handle)));
```

Comment:
Same logic need to be added to line 380 (get_known_avatar_tokens) as it doesn't seem to be deprecated.
Eg check hash - if exists fill the token, otherwise try to obtain presence, further code will work on presence if it exists and will skip it if token was filled (and presence remains null)